pkg/controller/migplan/validation.go

```diff
@@ -759,6 +771,115 @@ func (r ReconcileMigPlan) validatePvSelections(plan *migapi.MigPlan) error {
 	return nil
 }
 
+// Validate proxy secrets. Should only exist 1 or none
+func (r ReconcileMigPlan) validateRegistryProxySecret(plan *migapi.MigPlan) error {
+	misconfiguredClusters := 0
+	if plan.Status.HasAnyCondition(Suspended) {
+		return nil
+	}
+
+	list := kapi.SecretList{}
+	selector := k8sLabels.SelectorFromSet(map[string]string{
+		"migration-proxy-config": "true",
+	})
+
```

Comment:
Following the pattern of other validations I anticipated:

Types:  _like_
- SourceClusterProxyMisconfigured
- DestinationClusterProxyMisconfigured

Reasons: _like_
- Conflict
- KeyNotFound

One method for each (source, destionation).

Each method could check for (using source cluster example):
```
if plan.Status.HasAnyCondition(Suspended, InvalidSourceClusterRef, SourceClusterNotReady) {
      return nil
}
...
if len(list.Items) > 1 {
    plan.Status.SetCondition(migapi.Condition{
        Type:     SourceClusterProxyMisconfigured,
        Status:   True,
        Reason:   Conflict,
        Category: Critical,
        Message:  SourceClusterProxyMisconfiguredMessage,
    })
    return nil
}
if !r.validateProxySecret(list.Items[0]) {
        plan.Status.SetCondition(migapi.Condition{
        Type:     SourceClusterProxyMisconfigured,
        Status:   True,
        Reason:   KeyNotFound,
        Category: Critical,
        Message:  SourceClusterProxyMisconfiguredMessage,
    })
}
```

```
func (r *ReconcileMigPlan) validateProxySecret(secret *kapi.Secret) bool {
    fields := []string{
        settings.HttpProxy,
        settings.HttpsProxy,
        settings.NoProxy,
    }
    for _, key := range fields {
        if _, found := secret.Data[key]; found {
            return true
        }
    }

    return false
}
```

Reply:
Used a lot of this in the commit. Thank you very much for the thorough explanation.